src/App.js

```diff
@@ -1,13 +1,68 @@
 import React from 'react';
+import TodoList from './components/TodoComponents/TodoList';
+import TodoForm from './components/TodoComponents/TodoForm';
+import { EventEmitter } from './utils.js'
+import styles from './components/TodoComponents/Todo.css'
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor(props) {
+    super(props);
+    this.state = {
+      todo: [],
+      inputText: ''
+    }
+  EventEmitter.subscribe('toggleTodo', (id) => this.toggleTodo(id))
+	EventEmitter.subscribe('addTodo', (event) => this.addTodo(event))
+	EventEmitter.subscribe('clear', (event) => this.clear(event))
+  }
+
+  changeHandler = event => {
+    this.setState({
+      inputText: event.target.value
+    })
+  }
+
+  addTodo = event => {
+    event.preventDefault();
+    this.setState({
+      todo: [
+        ...this.state.todo,
+        {
+          task: this.state.inputText,
+          id: Date.now(),
+          completed: false
+        }
+      ],
+      inputText: ''
+    })
+  }
+
+  toggleTodo = id => {
+    this.setState({
+      todo: this.state.todo.map(
+        item => item.id === id ? Object.assign(item, { completed: !item.completed }) : item
```

Comment:
Nice use of `Object.assign`, but all the cool kids are using the spread operator
`item => item.id === id ? {...item, completed: !item.completed} : item`